lib/camp.js

```diff
@@ -739,18 +739,33 @@ function Path(matcher) {
 }
 
 // Path unit.
-function pathLikeUnit(serverMethod, statusCode) {
-  statusCode = statusCode || 200;
+function pathLikeUnit(serverMethod, defaultStatusCode, httpMethods) {
+  defaultStatusCode = defaultStatusCode || 200;
+  httpMethods = httpMethods || [];
   return function unit(server) {
     var paths = [];
     var callbacks = [];
+    var methodCallbacks = {};
 
-    var addPath = function (path, callback) {
+    var addCallback = function (path, callback) {
       paths.push(new Path('/' + path));
       callbacks.push(callback);
     };
+    server[serverMethod] = addCallback;
+
+    var addMethodCallback = function (method, path, callback) {
+      method = method.toLowerCase();
+      if (!methodCallbacks[path]) {
+        methodCallbacks[path] = {};
+        addCallback(path, methodCallbacks[path]);
+      }
+      methodCallbacks[path][method] = callback;
+    };
+    httpMethods.forEach(function (method) {
+      method = method.toLowerCase();
+      server[method] = addMethodCallback.bind(null, method);
```

Comment:
`method` must be lowercased for `server[method.toLowerCase()]`, but it is not needed elsewhere.